From 6e2c3d5c05223730711872319989a5929d49cef5 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:36:08 -0400
Subject: [alsa] hda: Add pin quirk for Dell XPS 15

Message-id: <140972823573-350-patchset-jkysela@redhat.com>
Patchwork-id: 91046
O-Subject: [RHEL 7.1 PATCH 349/442] ALSA: hda - Add pin quirk for Dell XPS 15
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Two bug reporters with Dell XPS 15 report that they need to use the
dell-headset-multi model to get the headset mic working.

The two bug reporters have different PCI SSID (1028:05fd and 1028:05fe)
but this pin quirk matches both.

BugLink: https://bugs.launchpad.net/bugs/1331915
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 7a52cd79fa2a445f6a57e9122d579883e07214fb)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 4852120..c70d43f 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -6237,6 +6237,27 @@ static const struct snd_hda_pin_quirk alc662_pin_fixup_tbl[] = {
   },
   .value = ALC668_FIXUP_AUTO_MUTE,
  },
+ {
+  .codec = 0x10ec0668,
+  .subvendor = 0x1028,
+#ifdef CONFIG_SND_DEBUG_VERBOSE
+  .name = "Dell XPS 15",
+#endif
+  .pins = (const struct hda_pintbl[]) {
+   {0x12, 0x90a60130},
+   {0x14, 0x90170110},
+   {0x15, 0x0321101f},
+   {0x16, 0x40000000},
+   {0x18, 0x411111f0},
+   {0x19, 0x411111f0},
+   {0x1a, 0x411111f0},
+   {0x1b, 0x411111f0},
+   {0x1d, 0x40d6832d},
+   {0x1e, 0x411111f0},
+   {0x1f, 0x411111f0},
+  },
+  .value = ALC668_FIXUP_AUTO_MUTE,
+ },
  {}
 };
 
-- 
1.7.1